group_project_v2/mixins.py

```diff
@@ -160,22 +327,18 @@ def workgroup(self):
         workgroup = self._get_workgroup(self.project_api, self.user_id, self.course_id)
         return workgroup if workgroup else self.FALLBACK_WORKGROUP
 
-    @staticmethod
-    @memoize_with_expiration()
-    def _get_workgroup(project_api, user_id, course_id):
+    def _get_workgroup(self, project_api, user_id, course_id):
         """
         :rtype: WorkgroupDetails
```

Comment:
@e-kolpakov  If i drop ``@staticmethod`` and leave ``@memoize_with_expiration()`` we'll end up with a slightly different semantics: as in case of ``@staticmethod`` memoisatnon will be type wide, and in ``@memoize_with_expiration()`` instance wide (which may be undesirable). 

So if thats OK with you, I'll: 

* Merge this to ``workgroup`` property. 
* Memoize ``get_user_roles_for_course``